### ISSUE TITLE
Let transpose ignore rests

### ISSUE THREAD

**Issue by jamesliddle** [wbsoft/frescobaldi#799]
I usually write rests in the form "b2\rest" to precisely specify their vertical position. When I transpose a piece with these rests, it moves the rests along with all the other pitches, and then I have to adjust them manually to put them back. It would be nice to have an option to transpose while leaving the rests in place.

**Comment by wbsoft** [comment]
Changing pitches displayed as `\rest` should become an option. Normally when notes move, it is sensible that "pitched" rests move too.

**Comment by jamesliddle** [comment]
Right. That should probably be the default behavior. I would just like an option to exclude pitched rests.

**Comment by marnen** [comment]
Also, when pitched rests move, they should normally be constrained to a line, not a space, no?

**Comment by jamesliddle** [comment]
I agree as a default, but, as with excluding pitched rests from transposition, this constraint should probably be a separate setting that can be toggled on or off.

**Comment by marnen** [comment]
Probably so (although I can't imagine *ever* wanting a pitched rest on a space).

**Comment by jamesliddle** [comment]
Sometimes smaller rests (quarter, eighth, and lower) can look fine centered on a space, and it can be helpful for tweaking placement in particularly dense scores.

**Comment by marnen** [comment]
Actually, that's poor engraving practice AFAIK. An eighth or smaller rest in a space will have the hook collide with the line, which is unattractive and hard to read. Quarter rests are still legible but just look wrong when placed at a space pitch. 

Space rests outside the staff are a different matter, now that I think about it, because there aren't lines to collide with. 

So while I agree that the behavior of pitched rests when transposed should be configurable, I think the default should probably be to constrain to spaces. :)

**Comment by marnen** [comment]
Er, I mean constrain to lines. D'oh!

**Comment by jamesliddle** [comment]
Agreed.
